From 1ee61a6dee4dedd071ea82535701d5ecf2ac6467 Mon Sep 17 00:00:00 2001
From: Stefan Kalscheuer <stefan@stklcode.de>
Date: Fri, 15 Aug 2025 15:00:43 +0200
Subject: [PATCH] arm64: dts: marvell: update LED and partition labels for
 GL-MV1000

Migrate LED labels from label to function/color.
Adjust partition labels and regions for OpenWRT compatibility.

Signed-off-by: Stefan Kalscheuer <stefan@stklcode.de>
---
 .../dts/marvell/armada-3720-gl-mv1000.dts     | 23 ++++++++++++++-----
 1 file changed, 17 insertions(+), 6 deletions(-)

--- a/arch/arm64/boot/dts/marvell/armada-3720-gl-mv1000.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-gl-mv1000.dts
@@ -3,6 +3,7 @@
 /dts-v1/;
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/input/input.h>
+#include <dt-bindings/leds/common.h>
 #include "armada-372x.dtsi"
 
 / {
@@ -63,12 +64,14 @@
 		};
 
 		led-wan {
-			label = "green:wan";
+			function = LED_FUNCTION_WAN;
+			color = <LED_COLOR_ID_GREEN>;
 			gpios = <&gpionb 12 GPIO_ACTIVE_LOW>;
 		};
 
 		led_power: led-power {
-			label = "green:power";
+			function = LED_FUNCTION_POWER;
+			color = <LED_COLOR_ID_GREEN>;
 			gpios = <&gpionb 13 GPIO_ACTIVE_LOW>;
 			default-state = "on";
 		};
@@ -89,8 +92,9 @@
 			#size-cells = <1>;
 
 			partition@0 {
-				label = "firmware";
+				label = "u-boot";
 				reg = <0 0xf0000>;
+				read-only;
 			};
 
 			partition@f0000 {
@@ -105,14 +109,21 @@
 			};
 
 			partition@100000 {
-				label = "dtb";
+				label = "gl-firmware-dtb";
 				reg = <0x100000 0x10000>;
 				read-only;
 			};
 
 			partition@110000 {
-				label = "rescue";
-				reg = <0x110000 0x1000000>;
+				label = "gl-firmware";
+				reg = <0x110000 0xef0000>;
+				read-only;
+			};
+
+			partition@ef0000 {
+				label = "gl-firmware-jffs2";
+				reg = <0xef0000 0x110000>;
+				read-only;
 			};
 		};
 	};
